packages/react-hooks/src/useEntryPosition.js

```diff
@@ -97,7 +98,7 @@ export const useEntryPosition = () => {
 
     // 🧹 Clean up when we're done.
     return () => document.removeEventListener('scroll', onScroll);
-  }, [onScroll, target, entryObserver, intersectionRatio]); // eslint-disable-line react-hooks/exhaustive-deps
+  }, [onScroll, target, entryObserver, intersectionRatio, handleScroll]);
```

Comment:
Oddly enough, keeping `handleScroll` out of the `useEffect` dependencies list allows it to be more performant and handle strange edge cases with React's state updating while a user scrolls slowly through an intersection observer threshold.

Reply:
Reflecting back on it, I should have added a comment 😅